python/tank_vendor/shotgun_authentication/user.py

```diff
@@ -0,0 +1,392 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Defines the supported types of authentication methods with Shotgun. You can
+either authenticate with a session token with the SessionUser class or with an
+api key with the ScriptUser class. This module is meant to be used internally.
+"""
+
+import pickle
+from .shotgun_wrapper import ShotgunWrapper
+
+from . import session_cache
+from .errors import CachingVolatileUserException
+
+
+_shotgun_instance_factory = ShotgunWrapper
```

Comment:
Thanks for the helpful comment below! (but maybe should be a comment and not a docstring for consistency?)

Should it be called factory though? It's just an alias for the `ShotgunWrapper` class. Factories normally produce instances of things and are usually methods - e.g. `tk.sgtk_from_path()`.